### PR TITLE
[NRLF-577] remove parameters for count

### PR DESCRIPTION
**Pull request by axelkrastek1-nhs**
Count End Point has incorrect parameters
https://nhsd-jira.digital.nhs.uk/browse/NRLF-577

### CODE REVIEW

File: feature_tests/features/consumer/consumer-countDocumentPointer-failure.feature [hunk 1]
```diff
@@ -111,39 +111,12 @@ Feature: Consumer Search Failure scenarios
...
     And the status is 400
     And the response is an OperationOutcome according to the OUTCOME template with the below values
-      | property          | value                                                                 |
-      | issue_type        | processing                                                            |
-      | issue_level       | error                                                                 |
-      | issue_code        | VALIDATION_ERROR                                                      |
-      | issue_description | A parameter or value has resulted in a validation error               |
-      | message           | ConsumerRequestParams validation failure - Invalid subject:identifier |
-
-  Scenario: Search rejects request with type system they are not allowed to use
```
Comment (andrewrae5-nhs) [inline feature_tests/features/consumer/consumer-countDocumentPointer-failure.feature:121]: Is there a reason why this scenario has been removed?

File: feature_tests/features/consumer/consumer-countDocumentPointer-success.feature [hunk 2]
```diff
@@ -143,67 +143,9 @@ Feature: Consumer Search Success scenarios
...
       | contentType | application/pdf                |
       | url         | https://example.org/my-doc.pdf |
-    When Consumer "Yorkshire Ambulance Service" counts Document References with query parameters:
-      | property             | value                                               |
-      | subject:identifier   | https://fhir.nhs.uk/Id/nhs-number\|9278693472       |
-      | custodian:identifier | https://fhir.nhs.uk/Id/ods-organization-code\|8FW23 |
-    Then the operation is successful
-    And the response has 1 total
-
-  Scenario: Successfully searches for all documents by type and nhs number for a consumer
```
Comment (andrewrae5-nhs) [inline feature_tests/features/consumer/consumer-countDocumentPointer-success.feature:153]: Again, why is this feature test and the one below removed?

**Review by andrewrae5-nhs — APPROVED** [review]
Looks good, just a few comments about removing gherkin tests.

File: feature_tests/features/consumer/consumer-countDocumentPointer-failure.feature (hunk 1 above)
Comment (axelkrastek1-nhs) [reply feature_tests/features/consumer/consumer-countDocumentPointer-failure.feature:121]: Type is no longer accepted as a parameter

File: feature_tests/features/consumer/consumer-countDocumentPointer-success.feature (hunk 2 above)
Comment (axelkrastek1-nhs) [reply feature_tests/features/consumer/consumer-countDocumentPointer-success.feature:153]: Type is no longer allowed

**Review by jaklinger — APPROVED** [review]

**Pull request merged**